src/crypto.rs

```diff
@@ -556,6 +569,58 @@ impl TryFrom<&serde_cbor::Value> for COSEKey {
     }
 }
 
+impl TryFrom<&X509PublicKey> for COSEKey {
+    type Error = WebauthnError;
+    fn try_from(cert: &X509PublicKey) -> Result<COSEKey, Self::Error> {
+        let ec_key = cert
+            .pubk
+            .public_key()
+            .and_then(|pk| pk.ec_key())
```

Comment:
And this should be moved into the match cert.t and we only get ec_key and the ec_group inside the match arm.

Reply:
Once it's under the match arm, is it okay to call it ec_key (since we do know it's type)?

Reply:
Yes, once it's in the match we can call it that.